metrics/metrics/src/main/java/io/helidon/metrics/ExponentiallyDecayingReservoir.java

```diff
@@ -135,6 +102,42 @@ private long computeCurrentTimeInSeconds() {
         this.nextScaleTime = new AtomicLong(clock.nanoTick() + RESCALE_THRESHOLD);
     }
 
+    static void init() {
+        if (EXECUTOR_UPDATER_CONFIGURED.compareAndSet(false, true)) {
+            ScheduledExecutorService result = Executors.newSingleThreadScheduledExecutor();
+            result.scheduleAtFixedRate(ExponentiallyDecayingReservoir::updateCurrentTimeInSecondsForAllReservoirs,
+                                       CURRENT_TIME_IN_SECONDS_UPDATE_INTERVAL_MS,
+                                       CURRENT_TIME_IN_SECONDS_UPDATE_INTERVAL_MS, TimeUnit.MILLISECONDS);
+            currentTimeUpdaterExecutorService = result;
+        }
+    }
+
+    // we only expect a single instance of MetricSupport registered
+    // with webserver, so we can stop the executor when the associated webserver stops
+    static void onServerShutdown() {
+        ScheduledExecutorService service = currentTimeUpdaterExecutorService;
+        if (EXECUTOR_UPDATER_CONFIGURED.compareAndSet(true, false)) {
+            try {
+                boolean stoppedNormally =
+                        service.awaitTermination(CURRENT_TIME_IN_SECONDS_UPDATE_INTERVAL_MS * 10,
+                                                 TimeUnit.MILLISECONDS);
+                if (!stoppedNormally) {
+                    LOGGER.log(Level.WARNING, "Shutdown of current time updater timed out; continuing");
+                }
+            } catch (InterruptedException e) {
+                LOGGER.log(Level.WARNING, "InterruptedException caught while stopping the current time updater; continuing");
+            }
+        }
+    }
```

Comment:
Also, since I made the original change to introduce this new executor, I've reminded myself that `HelidonConcurrentGauge` has its own `currentTimeMinute` method which could be streamlined if it used this same current-time-in-seconds mechanism. So it might make sense to refactor this into a separate package-private class which both `HelidonConcurrentGauge` and `ExponentiallyDecayingReservoir` would use.

Reply:
That should probably be its own separate issue and PR, but I wanted to mention it here.